packages/types/src/payment/processors.ts

```diff
@@ -0,0 +1,310 @@
+  /**
+   * The customer's email.
+   */
+  email: string
+  /**
+   * The selected currency code, typically associated with the customer's cart.
+   */
+  currency_code: string
+  /**
+   * The payment's amount.
+   */
+  amount: number
+  /**
+   * The ID of the resource the payment is associated with. For example, the cart's ID.
+   */
+  resource_id: string
+  /**
+   * The customer associated with this payment.
+   */
+  customer?: Record<string, unknown> // TODO: type
+  /**
+   * The cart's context.
+   */
+  context: Record<string, unknown>
+  /**
+   * If the payment session hasn't been created or initiated yet, it'll be an empty object.
+   * If the payment session exists, it'll be the value of the payment session's `data` field.
+   */
+  paymentSessionData: Record<string, unknown>
+}
+
+/**
+ * @interface
+ *
+ * The response of operations on a payment.
+ */
+export type PaymentProcessorSessionResponse = {
+  /**
+   * Used to specify data that should be updated in the Medusa backend.
+   */
+  update_requests?: {
+    /**
+     * Specifies a new value of the `metadata` field of the customer associated with the payment.
+     */
+    customer_metadata?: Record<string, unknown>
+  }
+  /**
+   * The data to be stored in the `data` field of the Payment Session to be created.
+   * The `data` field is useful to hold any data required by the third-party provider to process the payment or retrieve its details at a later point.
+   */
+  session_data: Record<string, unknown>
+}
+
+export type PaymentProcessorAuthorizeResponse = {
+  /**
+   * The status of the payment, which will be stored in the payment session's `status` field.
+   */
+  status: PaymentSessionStatus
+  /**
+   * The `data` to be stored in the payment session's `data` field.
+   */
+  data: PaymentProcessorSessionResponse["session_data"]
+}
+
+export type PaymentSessionInput = {
+  payment_session_id?: string
+  provider_id: string
+
+  cart_id?: string
+  email: string
+
+  // cart:
+  //   | Cart
+  //   | {
+  //       context: Record<string, unknown>
+  //       id: string
+  //       email: string
+  //       shipping_address: Address | null
+  //       shipping_methods: ShippingMethod[]
+  //       billing_address?: Address | null
+  //     }
+
+  customer_id: string
+  // customer?: Customer | null
+
+  currency_code: string
+  amount: number
+  resource_id: string
+  paymentSessionData?: Record<string, unknown>
+}
+
+export type CreatePaymentInput = {
+  amount: number
+  currency_code: string
+  provider_id?: string
+  payment_session: PaymentSessionDTO
+  resource_id?: string
+}
+
+/**
+ * An object that is returned in case of an error.
+ */
+export interface PaymentProcessorError {
+  /**
+   * The error message
+   */
+  error: string
+  /**
+   * The error code.
+   */
+  code?: string
+  /**
+   * Any additional helpful details.
+   */
+  detail?: any
+}
+
+export interface IPaymentProcessor {
```

Comment:
todo: Update JSDoc to not be coupled to other concepts e.g. Region and Cart. The payment provider should be capable of powering payments in any context.